plugins/modules/network/routeros/routeros_api.py

```diff
@@ -0,0 +1,489 @@
+            self.return_result(True)
+        except Exception as e:
+            self.errors(e)
+
+    def api_update(self):
+        param = self.list_to_dic(self.update.split(' '))
+        if '.id' not in param.keys():
+            self.errors("missing '.id' for %s" % param)
+        try:
+            self.api_path.update(**param)
+            self.result['message'].append("updated: %s" % param)
+            self.return_result(True)
+        except Exception as e:
+            self.errors(e)
+
+    def api_query(self):
+        keys = {}
+        for k in self.query:
+            if 'id' in k and k != ".id":
+                self.errors("'%s' must be '.id'" % k)
+            keys[k] = Key(k)
+        try:
+            if self.where:
+                if len(self.where) < 3:
+                    self.errors("invalid syntax for 'WHERE %s'"
+                                % ' '.join(self.where))
+
+                where = []
+                if self.where[1] == '==':
+                    select = self.api_path.select(*keys).where(keys[self.where[0]] == self.where[2])
+                elif self.where[1] == '!=':
+                    select = self.api_path.select(*keys).where(keys[self.where[0]] != self.where[2])
+                elif self.where[1] == '>':
+                    select = self.api_path.select(*keys).where(keys[self.where[0]] > self.where[2])
+                elif self.where[1] == '<':
+                    select = self.api_path.select(*keys).where(keys[self.where[0]] < self.where[2])
+                else:
+                    self.errors("'%s' is not operator for 'where'"
+                                % self.where[1])
+                for row in select:
+                    self.result['message'].append(row)
+            else:
+                for row in self.api_path.select(*keys):
+                    self.result['message'].append(row)
+            if len(self.result['message']) < 1:
+                msg = "no results for '%s 'query' %s" % (' '.join(self.path),
+                                                         ' '.join(self.query))
+                if self.where:
+                    msg = msg + ' WHERE %s' % ' '.join(self.where)
+                self.result['message'].append(msg)
+            self.return_result(False)
+        except Exception as e:
+            self.errors(e)
+
+    def api_arbitrary(self):
+        param = {}
+        self.arbitrary = self.arbitrary.split(' ')
+        arb_cmd = self.arbitrary[0]
+        if len(self.arbitrary) > 1:
+            param = self.list_to_dic(self.arbitrary[1:])
+        try:
+            arbitrary_result = self.api_path(arb_cmd, **param)
+            for i in arbitrary_result:
+                self.result['message'].append(i)
+            self.return_result(False)
+        except Exception as e:
+            self.errors(e)
+
+    def return_result(self, ch_status=False, status=True):
+        if not status:
+            self.module.fail_json(msg=self.result['message'])
```

Comment:
```suggestion
            self.module.fail_json(msg=to_native(self.result['message']))
```
for other `fail_json` cases as well, you can find it in the dev guide